src/main/java/io/neonbee/endpoint/odatav4/internal/olingo/OlingoEndpointHandler.java

```diff
@@ -43,245 +31,44 @@
-                            // the new routes! Might consider checking again with the Vert.x 4.0 release.
-                            routingContext.reroute(routingContext.request().uri());
-                        })));
-
-        return router;
-    }
-
-    private static Future<Void> refreshRouter(Vertx vertx, Router router, String basePath, UriConversion uriConversion,
-            AtomicReference<Map<String, EntityModel>> currentModels) {
-        return getSharedModels(vertx).compose(models -> {
-            if (models == currentModels.get()) {
-                return succeededFuture(); // no update needed
-            } else {
-                currentModels.set(models);
-            }
-
-            // before adding new routes, get a list of existing routes, to remove after the new routes have been added
-            List<Route> existingRoutes = router.getRoutes();
-
-            // Register new routes first, this will avoid downtimes of already existing services. Register the shortest
-            // routes last, this will lead to some routes like the empty namespace / to be registered last.
-            models.values().stream().flatMap(entityModel -> entityModel.getEdmxes().entrySet().stream())
-                    .map(entryFunction(
-                            (schemaNamespace, edmxModel) -> Map.entry(uriConversion.apply(schemaNamespace), edmxModel)))
-                    .sorted(Map.Entry.comparingByKey(Comparator.comparingInt(String::length).reversed()))
-                    .forEach(entryConsumer((uriPath, edmxModel) -> {
-                        router.route((uriPath.isEmpty() ? EMPTY : ("/" + uriPath)) + "/*").handler(create(edmxModel));
-                        LOGGER.info("Serving OData service endpoint for {} at {}{} ({} URI mapping)",
-                                edmxModel.getEdm().getEntityContainer().getNamespace(), basePath, uriPath,
-                                uriConversion.name().toLowerCase(Locale.getDefault()));
-                    }));
-
-            // remove any of the old routes, so the old models will stop serving
-            existingRoutes.forEach(Route::remove);
-
-            LOGGER.info("Refreshed OData endpoint router, populated {} models, removed {} existing routes",
-                    models.size(), existingRoutes.size());
-            return succeededFuture();
-        });
-    }
-
     /**
      * Convenience method as similar other Vert.x handler implementations (e.g. ErrorHandler)
      *
      * @param serviceMetadata The metadata of the service
-     * @return A ODataEndpointHandler instance
+     * @return A OlingoEndpointHandler instance
      */
-    public static ODataEndpointHandler create(ServiceMetadata serviceMetadata) {
-        return new ODataEndpointHandler(serviceMetadata);
+    public static OlingoEndpointHandler create(ServiceMetadata serviceMetadata) {
+        return new OlingoEndpointHandler(serviceMetadata);
     }
```

Comment:
This is something a recognized already a while ago, The reason why Vert.x has these convenience methods is, that their handlers are Interfaces which offers then concrete implementations with these methods. In this case OlingoEndpointHandler is already a concrete implementation and must be compared with "ErrorHandlerImpl" which doesn't offer a static create method.

I would  like to remove the static create method here.